test/test-cases/utils/snappi_utils.py

```diff
@@ -0,0 +1,62 @@
+def config_l1_properties(dataplane, usd_speed):
+    ly = dataplane.configuration.layer1.layer1(name="ly")[-1]
+    ly.port_names = [p.name for p in dataplane.configuration.ports]
+    val = getattr(ly , usd_speed )
+    setattr(ly , "speed", val)  
+    ly.ieee_media_defaults = False
+    ly.auto_negotiation.rs_fec = True
+    ly.auto_negotiation.link_training = False
+    ly.auto_negotiate = False
+
+def check_flow_tx_rx_frames_stats(dataplane, flow_name):
+    req = dataplane.api.metrics_request()
+    req.flow.flow_names = [flow_name]
+    flow_stats = dataplane.api.get_metrics(req)
+    print("statistics : {}".format(flow_stats))
+    frames_tx = sum([m.frames_tx for m in flow_stats.flow_metrics])
+    frames_rx = sum([m.frames_rx for m in flow_stats.flow_metrics])
+    return frames_tx == frames_rx
+
+def check_port_tx_rx_frames_stats(dataplane, port_name):
+    req = dataplane.api.metrics_request()
+    req.port.port_names = [port_name]
+    req.port.column_names = [req.port.FRAMES_TX, req.port.FRAMES_RX]
+    port_stats = dataplane.api.get_metrics(req)
+    print("statistics : {}".format(port_stats))
+    frames_tx = sum([m.frames_tx for m in port_stats.port_metrics])
+    frames_rx = sum([m.frames_rx for m in port_stats.port_metrics])
+    return frames_tx == frames_rx
+
+def check_bgp_neighborship_established(dataplane):
+    req =dataplane.api.metrics_request()
+    req.bgpv4.column_names = ["session_state"]
+    results = dataplane.api.get_metrics(req)
+    ok = []
+    for r in results.bgpv4_metrics:
+        ok.append(r.session_state == "up")
+    return all(ok)
+
+def check_ping(dataplane, ip_obj_name, ip, ip_type="ipv4"):
+    req = dataplane.api.ping_request()
+    if ip_type == "ipv4":
+        p1 = req.endpoints.ipv4()
+    else:
+        p1 = req.endpoints.ipv6()
+    p1.src_name = ip_obj_name
+    p1.dst_ip   = ip
+    responses = dataplane.api.send_ping(req).responses
+    for resp in responses :
+        if resp.src_name == ip1.name and resp.dst_ip == ip:
+            return True
+    else:
+        return False
+
+def start_traffic(dataplane, flow_name=None):
```

Comment:
It looks like this function is almost like the embedded one: https://github.com/opencomputeproject/SAI-Challenger/blob/3f6e90c5f5283782abcc21261861d369752144f2/common/sai_dataplane/snappi/sai_snappi_dataplane.py#L134

The difference is in passing flow names. Maybe it would be good to extend an existing function with a single attribute.

Reply:
Yes.. But we need to start the traffic with flow name instead starting all.. And we don't want to touch sai_snappi_dataplane.py as it is not maintained by us. Further we need discussion if we can combinedly work on sai_snappi_dataplane.py